### ISSUE TITLE
SDL 2.28.3: Regression on the iBuffalo classic usb gamepad 

### ISSUE THREAD

**Issue by frranck** [libsdl-org/SDL#8341]
Hello, I noticed a regression with the "Buffalo classic usb gamepad" in [my game](https://github.com/Javanaise/mrboom-libretro) on OSX while using SDL 2.28.3.
I haven't tested other OS versions.
The code dealing with joysticks hasn't changed, some other pads are still working fine.
The game receives a SDL_JOYDEVICEADDED event but no other event after that.
The controller still works fine into the SDL2 Gamepad Tool. (It's named Retro controller there)

**Comment by frranck** [comment]
SDL2 Gamepad Tool is using an older version: SDL2 Gamepad Tool v1.2 by General Arcade (compiled with SDL version 2.0.7, DLL version 2.0.7)

**Comment by slouken** [comment]
This may be related to SDL's GCController support. Thanks for the report!

**Comment by frranck** [comment]
You mean SDL_GameController? It stopped working while the code was only using SDL_Joystick.

**Comment by slouken** [comment]
It's a bug in Apple's GCController support. You can comment out SDL_JOYSTICK_MFI in the build config header and disable it. I have an open discussion with Apple about this.

**Comment by slouken** [comment]
This should be fixed in https://github.com/libsdl-org/SDL/commit/dcd21d042fe417d88bf00b203521eb140828b882, thanks!

**Comment by frranck** [comment]
Thanks! will test when a version is released

**Comment by frranck** [comment]
Hello @slouken, I did a test using the SDL2 branch of the library, It's now working but the mapping is incorrect and the up and down buttons are inverted.
I'm not sure how to replace the mapping to test it, but I got something like that trying the controllermap tool.
`05000000ac0500000400000050f16d04,USB2-axis 8-button gamepad,platform:Mac OS X,crc:f8c2,a:b0,b:b1,x:b4,y:b5,back:b2,start:b3,leftshoulder:b6,rightshoulder:b7,dpup:+a1,dpdown:-a1,dpleft:-a0,dpright:+a0,`
and the pad shows as 
`GUID:0500c2f8ac0500000400000050f16d04 JoystickName:USB,2-axis 8-button gamepad ControllerName:USB2-axis 8-button gamepad`
And hint on how to to test it? 
Shouldn't the mapping currently being used coming from SDL_gamecontrollerdb.h?
